library/src/main/java/org/xmtp/android/library/Group.kt

```diff
@@ -136,4 +141,26 @@ class Group(val client: Client, private val libXMTPGroup: FfiGroup) {
             libXMTPGroup.listMembers().map { it.accountAddress }
         }
     }
+
+    fun streamMessages(): Flow<DecodedMessage> = callbackFlow {
+        val messageCallback = object : FfiMessageCallback {
+            override fun onMessage(message: FfiMessage) {
+                trySend(Message(client, message).decode())
+            }
+        }
+
+        val stream = libXMTPGroup.stream(messageCallback)
+        awaitClose { stream.end() }
+    }
+
+    fun streamDecryptedMessages(): Flow<DecryptedMessage> = callbackFlow {
+        val messageCallback = object : FfiMessageCallback {
+            override fun onMessage(message: FfiMessage) {
+                trySend(decrypt(Message(client, message)))
```

Comment:
I think we want to refer to what you're doing in here as decoding rather than decrypting. The messages all get decrypted in `libxmtp`.

Reply:
So we have two streaming methods one for decoding and one for decrypting. We need this stream decrypted messages for React Native custom content types. @nakajima could speak more to it.

Since Groups are conforming to the current Conversation model I have to use these methods. Once we do the refactor to Groups Dms and Conversation parent we might be able to get rid of some of these things.

Reply:
What is the intended difference between a `DecryptedMessage` and a `DecodedMessage`? When I trace the method calls from both streaming methods, the `DecryptedMessage` one ultimately gives `message.decode().encodedContent`, and the `DecodedMessage` one ultimately gives `EncodedContent.parseFrom(message.content)`

Reply:
I don't think there is going to be any difference in the V3 world but again I didn't follow as closely to the reasoning for custom contentTypes. I found this for context https://github.com/xmtp/xmtp-ios/pull/196